treerec/implementation.md

```diff
@@ -5,62 +5,81 @@
     and reorder the individual tables so that the the currently alive ones come first,
     and in the order they currently exist in the population.
     Also, node times have the current generation added to them, so they are in units
     of number of generations before the end of the simulation.
-
-5. On input from tables, we use the `INDIVIDUAL_ALIVE` flag to decide who is currently alive,
-    and use the `INDIVIDUAL_REMEMBERED` flag to decide which genomes to add to `remembered_genomes_`.
-    We also remove the individuals that are not REMEMBERED (and therefore only ALIVE)
-    from the individual table; and also remove all ALIVE flags from individuals.
-    `msp_node_id_` is set based on the nodes that point to that individual in the tables.
-    Also, node times have the current generation subtracted from them, so they are in units
-    of number of generations before the start of the simulation.
-
+    Note that if simplify is *not* performed before output, all the nodes since the last
+    simplify will be marked with the `NODE_IS_SAMPLE` flag, even those not alive. Many of
+    these may not have an associated individual. This is deliberate, as we have full
+    genomic information for them, even if they are not ALIVE.
+
+5. On input from tables, we use the `INDIVIDUAL_ALIVE` flag to decide who is currently
+    alive and therefore must be instantiated into a SLiM individual, and use the
+    `INDIVIDUAL_REMEMBERED` flag to decide which genomes to add to `remembered_genomes_`.
+    From the individual table, we remove all individuals apart from those that are marked
+    as either `INDIVIDUAL_REMEMBERED` or `INDIVIDUAL_RETAINED` (thus we remove from the
+    table individuals that were only exported in the first place because they were ALIVE
+    at the time of export). `msp_node_id_` is set based on the nodes that point to that
+    individual in the tables. Also, node times have the current generation subtracted
+    from them, so they are in units of number of generations before the start of the simulation.
 
 Internal state:
 
 - `tables.individuals`:
 
-    * who's in the table: REMEMBERED
-    * flags: everyone is REMEMBERED (never ALIVE)
+    * who's in the table: REMEMBERED and (if they have an existing node) RETAINED
+    * flags: everyone either REMEMBERED or RETAINED or both (never ALIVE)
 
 - `tables.nodes`:
 
     * NODE_IS_SAMPLE flag: **unused** by SLiM; but set if they are in
       REMEMBERED individuals, or if they're from a generation since the
       last simplify (or the current generation, if that just happened)
 
-    * individual column: if their individual is REMEMBERED
+    * individual column: if their individual is REMEMBERED or RETAINED
 
         - created by AddIndividualsToTable, called by RememberIndividuals
 
     * time: (-1) * birth generation.
 
-- `remembered_genomes_`: the node IDs of genomes whose individuals are REMBEMBERED
```

Comment:
"REMBEMBERED" :->

Reply:
Good spot, thanks. I'll push an updated version

Reply:
OIC - I did correct it! The typo was in the old version. Nothing to see here, move along :)

Reply:
No, you already fixed this typo.  I'm just amused at the typo itself.  :->